resources/certificate_binding.rb

```diff
@@ -18,14 +18,111 @@
 # limitations under the License.
 #
 
-actions :create, :delete
-default_action :create
+include Chef::Mixin::ShellOut
+include Chef::Mixin::PowershellOut
+include Windows::Helper
 
-attribute :cert_name, kind_of: String, name_attribute: true, required: true
-attribute :name_kind, kind_of: Symbol, equal_to: [:hash, :subject], default: :subject
-attribute :address, kind_of: String, default: '0.0.0.0'
-attribute :port, kind_of: Integer, default: 443
-attribute :app_id, kind_of: String, default: '{4dc3e181-e14b-4a21-b022-59fc669b0914}'
-attribute :store_name, kind_of: String, default: 'MY', regex: /^(?:MY|CA|ROOT)$/
+property :cert_name, String, name_property: true, required: true
+property :name_kind, Symbol, equal_to: [:hash, :subject], default: :subject
+property :address, String, default: '0.0.0.0'
+property :port, Integer, default: 443
+property :app_id, String, default: '{4dc3e181-e14b-4a21-b022-59fc669b0914}'
+property :store_name, String, default: 'MY', regex: /^(?:MY|CA|ROOT)$/
+property :exists, [true, false], desired_state: true
```

Comment:
Similar to the comment on the acl provider if we can avoid having an exists property that would be great. This can be done using just a helper and converge_by with a similar amount of logic